.github/workflows/open-issue-in-repo.yml

```diff
@@ -0,0 +1,136 @@
+        required: false
+        type: string
+      issue_title:
+        description: 'Title of the issue'
+        required: true
+        type: string
+      issue_body:
+        description: 'The body of the issue'
+        required: true
+        type: string
+    secrets:
+      FISHTOWN_BOT_PAT:
+        description: 'PAT to use for authentication'
+        required: true
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  issues: write # opens new issues
+  pull-requests: write # comments on PRs
+
+jobs:
+  prep:
+    runs-on: ubuntu-latest
+    outputs:
+      exists: ${{ steps.comment_check.outputs.exists }}
+
+    steps:
+
+      - name: "[DEBUG] Print variables"
+        id: print_variables
+        run: |
+          echo "Repository name to open the issue:      ${{ inputs.issue_repository }}"
+          echo "issue_title:                            ${{ inputs.issue_title}}"
+          echo "issue_body:                             ${{ inputs.issue_body }}"
+      
+      # this step uses the read permission from the GITHUB_TOKEN it inherits
+      - name: "Check for comment"
+        uses: peter-evans/find-comment@v2
+        id: pr_comment
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          comment-author: "github-actions[bot]"
+          body-includes: "Opened a new issue in ${{ inputs.issue_repository }}:"
+
+      - name: "Set if comment already exists"
+        id: comment_check
+        run: |
+          if [ '${{ steps.pr_comment.outputs.comment-body }}' = '' ]; then
+            echo "exists=false" >> $GITHUB_OUTPUT
+            title="Comment does not exist for this PR."
+            message="Opening a new issue in ${{ inputs.issue_repository }}."
+          else
+            echo "exists=true" >> $GITHUB_OUTPUT
+            title="Comment already exists for this PR indicating associated issue."
+            message="New issue will not be opened."
+          fi
+          echo "::notice $title::$message"
+
+  open_new_issue:
+    needs: [prep]
+    if: ${{ needs.prep.outputs.exists == 'false' }}
+    runs-on: ubuntu-latest
+    outputs:
+      issue_url: ${{ steps.open_issue.outputs.issue_url }}
+
+    steps:
+      - name: Check out ${{ inputs.issue_repository }}
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ inputs.issue_repository }}
+          token: ${{ secrets.FISHTOWN_BOT_PAT }}
+      
+      - name: "Set labels"
+        id: labels
+        run: |
+          if [ '${{ inputs.issue_labels }}' != '' ]; then
+            echo "labels=--label '${{ inputs.issue_labels }}'" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Open Issue
+        id: open_issue
+        run: |
+          issue=$(gh issue create --repo ${{ inputs.issue_repository }} \
+            --title "${{ inputs.issue_title }}" \
+            --body "${{ inputs.issue_body }}" \
+            --assignee ${{ github.actor }} ${{ steps.labels.outputs.labels }})
```

Comment:
What happens if the if statement in `labels` is false, and nothing is export to GITHUB_OUTPUT? Does this evaluate to an empty string? Or does it cause an exception?

Reply:
It's not intuitive but GitHub just treats it as empty.  It doesn't error if you refer to it and it's not created.